### PR TITLE
feat: Hamletcli automation provider

### PR DESCRIPTION
**Pull request by roleyfoley**
## Intent of Change

- New feature (non-breaking change which adds functionality)

## Description

Adds support for the hamletcli as an automation provider. This allows for the automation scripts to provide support for using the cli as the script caller

Adds support for running the automation commands locally using the hamletcli provider: 
- Support for using an existing CMDB tree instead of constructing one. The ROOT_DIR must be defined along with the providing USE_EXISTING_TREE to the constructTree.sh script 
- Support for using user defined profiles when the automation scripts require AWS access
- Support for using the users global git config when commit changes to the repo

## Motivation and Context

The motivation behind this change is to to allow the hamlet cli to manage the process of running our automation commands and centralises the dependencies into hamlet maintained components which can run across different providers

## How Has This Been Tested?

Tested locally

### Prerequisite PRs:

- None

### Dependent PRs:

- None

### Consumer Actions:

- None

### CODE REVIEW

File: automation/constructTree.sh [hunk 1]
```diff
@@ -46,14 +47,17 @@ EOF
 }
 
 # Parse options
-while getopts ":ac:hi:rx:" opt; do
+while getopts ":ac:ef:hi:rx:" opt; do
```
Comment (ml019) [inline automation/constructTree.sh:50]:
```suggestion
while getopts ":ac:ehi:rx:" opt; do
```

File: automation/constructTree.sh [hunk 2]
```diff
@@ -85,235 +89,247 @@ ACCOUNT_CONFIG_REFERENCE="${ACCOUNT_CONFIG_REFERENCE:-$ACCOUNT_CONFIG_REFERENCE_
...
+    #        TENANT_INFRASTRUCTURE_DIR="${BASE_DIR}/${TENANT}"
+    #        mkdir -p $(filePath "${TENANT_INFRASTRUCTURE_DIR}")
+    #        mv "${BASE_DIR_TEMP}" "${TENANT_INFRASTRUCTURE_DIR}"
+    #    fi
+
+    fi
+fi
+
+if [[ "${USE_EXISTING_TREE}" == "true" ]]; then
+    if [[ -n "${ROOT_DIR}" && -d "${ROOT_DIR}" ]]; then
```
Comment (ml019) [inline automation/constructTree.sh:328]: This logic seems back the front? It is reporting an error is the variable is defined and is a directory...

File: automation/constructTree.sh [hunk 3]
```diff
@@ -85,235 +89,247 @@ ACCOUNT_CONFIG_REFERENCE="${ACCOUNT_CONFIG_REFERENCE:-$ACCOUNT_CONFIG_REFERENCE_
...
+        fatal "ROOT_DIR: ${ROOT_DIR} - could not be found for exsting tree"
```
Comment (ml019) [inline automation/constructTree.sh:329]:
```suggestion
        fatal "ROOT_DIR: ${ROOT_DIR} - could not be found for existing tree"
```

**Review by ml019 — CHANGES_REQUESTED** [review]

File: automation/constructTree.sh (hunk 2 above)
Comment (roleyfoley) [reply automation/constructTree.sh:328]:
```suggestion
    if [[ -z "${ROOT_DIR}" || -d "${ROOT_DIR}" ]]; then
```

Comment (roleyfoley) [reply automation/constructTree.sh:328]: Yeah sorry that should have been the opposite. Suggested change above If you can check it @ml019

**Review by ml019 — APPROVED** [review]

**Pull request merged**